### PR TITLE
Exclude broken oshi release

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5370

### CODE REVIEW

File: instrumentation/oshi/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -15,9 +17,13 @@ dependencies {
 
   compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")
 
-  library("com.github.oshi:oshi-core:5.3.1")
+  compileOnly("com.github.oshi:oshi-core:5.3.1")
+  testImplementation("com.github.oshi:oshi-core:5.3.1")
```
Comment (trask) [inline instrumentation/oshi/javaagent/build.gradle.kts:21]: why this change?

Comment (laurit) [reply instrumentation/oshi/javaagent/build.gradle.kts:21]: when using `library` it still attempted to parse the broken pom. Here is the failure with `library` https://github.com/open-telemetry/opentelemetry-java-instrumentation/runs/5185602614?check_suite_focus=true Anyway I expect that `library` and `latestDepTestLibrary` changes in this pr can be reverted tomorrow.

**Review by trask — APPROVED** [review]
